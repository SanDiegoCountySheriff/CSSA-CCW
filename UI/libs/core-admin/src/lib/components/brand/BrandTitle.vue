<template>
  <TextInput
    :label="'Brand Title '"
    :target="'brandTitle'"
    :rules="[v => !!v || 'Brand Title is required']"
    :name="'brandTitle'"
  />
</template>

<script setup lang="ts">
import TextInput from '@shared-ui/components/inputs/TextInput.vue';
</script>
