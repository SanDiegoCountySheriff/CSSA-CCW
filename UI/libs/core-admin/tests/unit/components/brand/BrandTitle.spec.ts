import BrandTitle from '@core-admin/components/brand/BrandTitle.vue';
import { createLocalVue, mount } from '@vue/test-utils';
import Vuetify from 'vuetify';

const localVue = createLocalVue();

describe('BrandTitle', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('Should render logo label', () => {
    const wrapper = mount(BrandTitle, {
      localVue,
      vuetify,
      mocks: {
        $t: text => {
          text;
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('.v-input').exists()).toBe(true);
  });
});
