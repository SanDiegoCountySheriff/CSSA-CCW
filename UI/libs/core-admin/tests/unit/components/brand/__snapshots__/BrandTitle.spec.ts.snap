// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandTitle Should render logo label 1`] = `
<div class="v-input theme--light v-text-field">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-text-field__slot"><input name="brandTitle" id="input-3" type="text"></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
